util/types/src/core/header_digest.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{core::views::HeaderView, packed, prelude::*, U256};
+use ckb_hash::new_blake2b;
+use ckb_merkle_mountain_range::{MerkleElem, Result as MMRResult};
+
+#[derive(Clone, Debug)]
+pub struct HeaderDigest {
+    data: packed::HeaderDigest,
+}
+
+impl HeaderDigest {
+    pub fn data(&self) -> &packed::HeaderDigest {
+        &self.data
+    }
+}
+
+impl From<packed::HeaderDigest> for HeaderDigest {
+    fn from(data: packed::HeaderDigest) -> Self {
+        HeaderDigest { data }
+    }
+}
+
+impl From<HeaderView> for HeaderDigest {
+    fn from(header_view: HeaderView) -> Self {
+        let data = packed::HeaderDigest::new_builder()
+            .hash(header_view.hash())
+            .total_difficulty(header_view.difficulty().pack())
+            .build();
+        HeaderDigest { data }
+    }
+}
+
+impl PartialEq for HeaderDigest {
+    fn eq(&self, other: &Self) -> bool {
+        self.data().as_slice().eq(other.data().as_slice())
+    }
+}
```

Comment:
Better use macro `impl_std_cmp_eq_and_hash` in `util/types/src/extension/std_traits.rs` ?